src/app/qgsclipboard.cpp

```diff
@@ -154,22 +154,38 @@ void QgsClipboard::setSystemClipboard()
   // that just the next call to systemClipboardChanged() should be ignored
   mIgnoreNextSystemClipboardChange = true;
 
-  QString textCopy = generateClipboardText();
-
   QClipboard *cb = QApplication::clipboard();
 
   // Copy text into the clipboard
+  QString textCopy = generateClipboardText();
+  QMimeData *m = new QMimeData();
+  m->setText( textCopy );
+
+  QgsSettings settings;
+  CopyFormat format = AttributesWithWKT;
+  if ( settings.contains( QStringLiteral( "/qgis/copyFeatureFormat" ) ) )
+  {
+    format = static_cast< CopyFormat >( settings.value( QStringLiteral( "qgis/copyFeatureFormat" ), true ).toInt() );
+  }
+
+  if ( format == AttributesWithWKT && mFeatureClipboard.count() < 1000 )
```

Comment:
Can you change this to a switch please? (It'll be safer if additional formats are added in future - a warning will be thrown and the developer will need to consider this code block too)